x/dclauth/types/message_approve_add_account_test.go

```diff
@@ -1,42 +1,59 @@
 package types
 
-/* TODO issue 99
 import (
 	"testing"
 
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/stretchr/testify/require"
-	"github.com/zigbee-alliance/distributed-compliance-ledger/testutil/sample"
+	testconstants "github.com/zigbee-alliance/distributed-compliance-ledger/integration_tests/constants"
 )
 
-func TestMsgApproveAddAccount_ValidateBasic(t *testing.T) {
+func TestNewMsgApproveAddAccount(t *testing.T) {
+	msg := NewMsgApproveAddAccount(testconstants.Signer, testconstants.Address1, testconstants.Info)
+
+	require.Equal(t, msg.Route(), RouterKey)
+	require.Equal(t, msg.Type(), "approve_add_account")
+	require.Equal(t, msg.GetSigners(), []sdk.AccAddress{testconstants.Signer})
+}
+
+func TestValidateMsgApproveAddAccount(t *testing.T) {
```

Comment:
It would be great to take into account an expected error for negative tests (valid=false). Also it makes sense to split negative (valid=false, error expected) and positive (valid message) tests.
Please look at https://github.com/zigbee-alliance/distributed-compliance-ledger/blob/master/x/model/types/messages_model_test.go as a reference.

Reply:
Done! ✅

Reply:
It looks like there are still no checks for expected error messages in the negative tests...